documentation/modules/ROOT/pages/ai/embeddings.adoc

```diff
@@ -0,0 +1,187 @@
+:page-aliases: ai/embeddings.adoc
+// Category: debezium-using
+// Type: assembly
+// ModuleID: embeddings-transformation
+// Title: Embeddings Transformation
+[id="embeddings-transformation"]
+= Embeddings Transformation
+ifdef::community[]
+:toc:
+:toc-placement: macro
+:linkattrs:
+:icons: font
+:source-highlighter: highlight.js
+
+toc::[]
+endif::community[]
+
+In large language models, or in natural language processing in general, one of the important tasks is conversion of the text to the numerical representation.
+One way how to do that is so-called text embeddings.
+It's a process of conversion of a text to a high-dimensional numerical vector.
+Representing text as a vector allows to do semantic similarity search and other advanced operations.
+
+{prodname} offers build-in feature which transforms specified field into embedding and add this text embedding into the record.
+Embedding inference is done by the selected model served by configured provider.
+{prodname} offers several embeddings model providers.
+
+Technically it's implemented as a  link:{link-kafka-docs}/#connect_transforms[single message transformation] (SMT) and under the hood it uses link:https://docs.langchain4j.dev/[langchain4j] framework to interact with embeddings providers.
+
+== Behavior
+
+Embeddings transformation takes specified record field and passes it to the configured embedding model for inference, i.e. creating an embedding from the text contained in this field.
+Resulting embedding is appended to the record.
+The source field is still part of the record.
+
+The source field has to be a string field.
+Embedding field is a vector of float numbers.
```

Comment:
I'd recommend to highlight that these are flot 16 numbers and also refer to Debezium's logical `io.debezium.data.FloatVector` type.

Reply:
I think they are float-32